File: core/src/main/java/reader/ecos/org/core/io/files/ZipNavigatorService.java
package reader.ecos.org.core.io.files;

import reader.ecos.org.core.io.files.exceptions.DoNoExistsExcepiton;

@SuppressWarnings("unused")
public interface ZipNavigatorService {
    FileCollection getFilesInside(ZipFile file,Path path) throws DoNoExistsExcepiton;
    DirectoryCollection getDirectoriesInside(ZipFile file,Path path);
}

File: files/src/main/kotlin/org/ecos/core/files/ZipNavigatorService.kt
package org.ecos.core.files

import org.ecos.reader.core.exceptions.DoNoExistsException

import org.ecos.reader.core.exceptions.UnknownError

interface ZipNavigatorService {
    @Throws(DoNoExistsException::class, UnknownError::class)
    fun getFilesInside(file: ZipFile): FileCollection
}